packages/application-config/src/process-config.ts

```diff
@@ -0,0 +1,108 @@
+import { JSONSchemaForCustomApplicationConfigurationFiles } from './schema';
+import type { ApplicationConfig, CloudIdentifier } from './types';
+
+// Loads the configuration file and parse the environment and header values.
+// Most of the resulting values are inferred from the config.
+import loadConfig from './load-config';
+import validateConfig from './validate-config';
+import {
+  mapCloudIdentifierToApiUrl,
+  getUniqueValues,
+  getIsProd,
+  getOrThrow,
+} from './utils';
+import substituteEnvVariablePlaceholders from './substitute-env-variable-placeholders';
+
+type ProcessConfigOptions = {
+  disableCache?: boolean;
+  processEnv?: NodeJS.ProcessEnv;
+};
+
+const developmentConfig: JSONSchemaForCustomApplicationConfigurationFiles['env']['production'] = {
+  url: 'http://localhost:3001',
```

Comment:
This assumes that the development URL of the app is always `http://localhost:3001`.

I can't think of a case where this value can be different, and therefore it's pointless to ask the user to provide this value.

If this needs to be configurable, we can refactor the implementation a bit to support that.

Reply:
I agree.